text/0000-add-funding-uri.md

```diff
@@ -0,0 +1,141 @@
+  }
+end
+```
+
+`funding_uri` must be a valid URI, which `metadata` will already enforce.
+
+After a user runs `bundle update` or `bundle install` on the command line, a message pops up with some additional information:
+
+```
+Updating files in vendor/cache
+  * byebug-11.1.0.gem
+  * ffi-1.12.1.gem
+  * mercenary-0.4.0.gem
+  * unicode-display_width-1.6.1.gem
+Removing outdated .gem files from vendor/cache
+  * unicode-display_width-1.6.0.gem
+  * byebug-11.0.1.gem
+  * mercenary-0.3.6.gem
+  * ffi-1.11.3.gem
+Bundle updated!
+```
+
+This RFC proposes printing out more information pertaining to the gems which have defined `funding_uri`:
+
+```
+Updating files in vendor/cache
+  * byebug-11.1.0.gem
+  * ffi-1.12.1.gem
+  * mercenary-0.4.0.gem
+  * unicode-display_width-1.6.1.gem
+Removing outdated .gem files from vendor/cache
+  * unicode-display_width-1.6.0.gem
+  * byebug-11.0.1.gem
+  * mercenary-0.3.6.gem
+  * ffi-1.11.3.gem
+Bundle updated!
+
+4 gems you depend on are looking for funding
+  Run `bundle fund` for details
+```
+
+This information is parsed out of each gem's `funding_uri` key. The user now has a choice: move on with their day, or, overwhelmed by compassion and curiosity, run `bundle fund`. If they choose the latter option, the following information is printed out:
+
+```
+$ bundle fund
+  * $GEM_NAME ($GEM_VERSION)
+	Funding: <$FUNDING_URL>
+```
+
+The name of the gem, and the funding page. That's it.
+
+As well, `bundle info GEM` should show this information as part of its printout:
+
+```
+$ bundle info GEM
+
+  * GEM (GEM_VERSION)
+	Homepage: GEM_HOMEPAGE
+	Funding:  GEM_FUNDING_PAGE
+```
+
+If there is no `funding_uri` key, no information will be displayed. If none of the gems have a `funding_uri` defined, the command should print a message out indicating that:
+
+```
+$ bundle fund
+None of the gems you depend on are looking for funding!
+```
+
+This is a backwards-compatible, additive change to bundler and the gem specification.
+
+# Reference-level explanation
+
+Right now, there is a very similar "print a message" gem attribute by way of `post_install_message`. A crucial difference between `post_install_message` and `funding_uri` is that `post_install_message` ought to be preserved for essential information. Sometimes, open source maintainers abuse this attribute by printing their own non-essential messages, to the annoyance of users. To this end, bundler added an `ignore_messages` configuration option, which silenced `post_install_message`s on an individual gem or a global level.
+
+I do not believe that displaying a single message indicating gems which need funding reaches this level of annoyance. Therefore, I do not expect `ignore_messages` to have an impact on `bundle fund`, or the `X gems are looking for funding` message printout, nor should an opt-out flag or configuration should be considered at this time.
+
+`bundle init` should also not add a placeholder for `funding_uri`.
```

Comment:
Do you mean `bundle gem` here? `bundle init` only just generates a `Gemfile`

Reply:
I did!